pkg/scheduler/framework/runtime/framework.go

```diff
@@ -258,6 +270,11 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
 		}
 		pluginConfig[name] = args[i].Args
 	}
+	outputProfile := config.KubeSchedulerProfile{
+		SchedulerName: f.profileName,
+		Plugins:       plugins,
```

Comment:
Line 307 changed the weight to 1 if it is 0, but I guess it should be fine. Maybe the weight==0 should be validated first in the future.

Reply:
It could be defaulted in `pkg/scheduler/apis/config/defaults.go`, but only for Score.

However, we still need to preserve the defaulting logic here too. The reason being that there are direct users of `NewFramework`, like cluster autoscaler.